foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_10_0/DeltaSparkYamlMigration.java

```diff
@@ -0,0 +1,105 @@
+package com.boozallen.aissemble.upgrade.migration.v1_10_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import com.boozallen.aissemble.upgrade.util.YamlUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.BatonException;
+import org.yaml.snakeyaml.error.YAMLException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+public class DeltaSparkYamlMigration extends AbstractAissembleMigration {
+    private static final Logger logger = LoggerFactory.getLogger(DeltaSparkYamlMigration.class);
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        if (file.getName().endsWith(".yaml")) {
+            try {
+                YamlUtils.YamlObject yaml = YamlUtils.loadYaml(file);
+                if (yaml.hasObject("sparkApp")) {
+                   yaml = yaml.getObject("sparkApp");
+                }
+                if (yaml.hasObject("spec")) {
+                   yaml = yaml.getObject("spec");
+                    if (yaml.hasObject("deps")) {
+                        yaml = yaml.getObject("deps");
```

Comment:
Maybe create an utility function to get the object from a path. e.g.: `YamlUtils.getObjectFromPath("sparkApp.spec.deps")` ?

Reply:
I just updated the `YamlObject` hasX/getX methods to accept full paths instead of single keys.  Prevents this nesting situation while not cluttering the `YamlObject` API.